src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,175 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidDataException;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+    private static final int ADULT_AGE = 18;
+    private static final int ANOTHER_ADULT_AGE = 25;
+    private static final int NEGATIVE_AGE = -10;
+    private static final int TEEN_AGE = 17;
+    private static final int ZERO_AGE = 0;
+
+    private static RegistrationService registrationService;
+    private User john;
+    private User jane;
+    private User maria;
+    private User george;
+    private String validPassword;
+    private String anotherValidPassword;
+    private String invalidPassword;
+    private String shortPassword;
+    private String validLogin;
+    private String anotherValidLogin;
+    private String thirdValidLogin;
+    private String invalidLogin;
+    private String shortLogin;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        john = new User();
+        jane = new User();
+        maria = new User();
+        george = new User();
+        validPassword = "qwerty";
+        anotherValidPassword = "qwe123rty";
+        invalidPassword = "abc12";
+        shortPassword = "ox";
+        validLogin = "UserName";
+        anotherValidLogin = "AcceptableLogin";
+        thirdValidLogin = "MyName";
+        invalidLogin = "NameX";
+        shortLogin = "Rw";
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_DifferentValidUsers_Ok() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        maria.setAge(ANOTHER_ADULT_AGE);
+        maria.setLogin(anotherValidLogin);
+        maria.setPassword(anotherValidPassword);
+        User actual = registrationService.register(john);
+        assertEquals(actual, john);
+        User nextUserToAdd = registrationService.register(maria);
+        assertEquals(nextUserToAdd, maria);
+    }
+
+    @Test
+    void register_underageUser_NotOk() {
```

Comment:
This test checks if underage users cannot register, which is good. However, it would be better to also include an edge case test for a user who is exactly 18 years old to ensure that the boundary condition is handled correctly.